src/ExadelMentorship.BusinessLogic/Services/Mail/EmailSender.cs

```diff
@@ -0,0 +1,56 @@
+﻿using ExadelMentorship.BusinessLogic.Interfaces;
+using ExadelMentorship.BusinessLogic.Models;
+using MailKit.Net.Smtp;
+using MailKit.Security;
+using Microsoft.Extensions.Options;
+using MimeKit;
+using MimeKit.Text;
+using System;
+
+namespace ExadelMentorship.BusinessLogic.Services.Mail
+{
+    public class EmailSender : IEmailSender
+    {
+        private EmailConfiguration _emailConfiguration;
+        public EmailSender(IOptions<EmailConfiguration> emailConfiguration)
+        {
+            _emailConfiguration = emailConfiguration.Value;
+        }
+        public void SendEmail(Message message)
+        {
+            var emailMessage = CreateEmailMessage(message);
+            Send(emailMessage);
+        }
+
+        private MimeMessage CreateEmailMessage(Message message)
+        {
+            var emailMessage = new MimeMessage();
+            emailMessage.From.Add(MailboxAddress.Parse(_emailConfiguration.From));
+            emailMessage.To.Add(MailboxAddress.Parse(message.To));
+            emailMessage.Subject = message.Subject;
+            emailMessage.Body = new TextPart(TextFormat.Text) { Text = message.Content };
+
+            return emailMessage; 
+        }
+
+        private void Send(MimeMessage mailMessage)
+        {
+            using var client = new SmtpClient();
+            try
+            {
+                client.Connect(_emailConfiguration.SmtpServer, _emailConfiguration.Port, true);
+                client.Authenticate(_emailConfiguration.UserName, _emailConfiguration.Password);
+
+                client.Send(mailMessage);
+            }
+            catch (Exception ex)
+            {
+                throw new AuthenticationException(ex.Message);
+            }
+            finally
+            {
+                client.Disconnect(true);
```

Comment:
`SmtpClient.Dispose` might already do this (and more). So you may need this finally block.

Reply:
done